designs/administrator/future-openshift/bulk-actions/index.md

```diff
@@ -0,0 +1,139 @@
+![infinite-scrolling-page-down](img/infinite-scrolling-page-down.png)
+
+### Filtering
+If a list view contains filtering, items that have already been selected will remain selected if filters are applied or cleared. 
+![filter-clear](img/filter-clear.png)
+![filter-update](img/filter-update.png)
+
+They may select additional items from the filtered list and once they do, the item count in the bulk selector will update to reflect the action. 
+![filter-select](img/filter-select.png)
+
+## Actions
+Only specific actions will be available to take on multiple resources. The default bulk actions are `Add Labels`, `Add Annotations` and `Delete Resources`. Additional actions may be added in future depending on the resource type. When taking actions on five resources or fewer, the modal will list out the resource names. When taking actions on more than five resources, the modal will simply list the number (e.g. “Labels for 25 resources.”).
+
+### Add Labels
+The `Add Labels` modal allows the user to add labels to multiple resources and notes that existing labels may be replaced if there are duplicates. Users can add labels in the textbox and click ‘Save’ to add the labels to the selected resources. The new labels will appear in the list and the items will remain selected after the modal closes.
+![add-labels-modal](img/add-labels-modal.png)
+
+### Add Annotations
+The `Add Annotations` modal allows the user to add annotations to multiple resources and notes that existing annotations may be replaced. Users can add annotations and click ‘Save’ to add the annotations to the selected resources. The items will remain selected after the modal closes.
+![add-annotations-modal](img/add-annotations-modal.png)
+
+### Delete Resources
+
+#### No failures
+
+The `Delete Resources` modal will list out the resources that are being chosen to be deleted and will be checked by default. There will be a bulk selector above the listed items that will showcase the item count and will act the same way that the bulk selector within the toolbar works. The user may change these checkboxes if desired, but this acts as a “second check” for them. The user must select “Delete” in order to begin the deletion process.
+![delete-resources-modal](img/delete-resources-modal.png)
+
+Once the deletion process begins, the messaging will be replaced with the below modal. The progress bar will track the completion progress. If the user decides to navigate away from the modal, either by selecting “Done”, the close “x” button, or navigating away from the page, the modal will close and they cannot reopen it. 
+![delete-resources-progress](img/delete-resources-progress.png)
+
+Assuming the deletion completes with no failures, the messaging will be repelaced with the below modal. 
+![delete-resources-complete](img/delete-resources-complete.png)
+
+Once the user navigates back to the list view, the items will be removed from the list (or in the process of being removed) and no longer selected.
```

Comment:
So if they exit the modal while the operation while it's still running, all the selected items will already be gone from the list? Or they'll disappear while it runs?

Related to the failures below, if they closed the modal and some failure occurs, would they be notified (the modal reopened?) I guess I ask since maybe if they start the operation, close the modal while its running, see the items gone from the list, and then it fails some time later (after they navigated away from the list,) they might never realize that the items weren't actually deleted.

Reply:
@itsptk Yeah this one is a pretty tough one. Since we don't have toast alerts, there was really no other way for us to notify the user of what is going on during the process. Basically, when they navigate away from the modal, the status within each of the items they took action on will be updating (if they were to examine the list view). Then they would not know about failures unless they noticed the items still in the list view.

That's a good idea to pop the modal back up even after the process is completed. Do we do that anywhere else?

Reply:
I personally can't think of any examples, hm.

If we definitely don't want to hold them in the modal while the process runs, it seems like it will just need to work like you described in the design. I didn't realize the status changed to 'updating' while they are in the process of being deleted, and do think that helps.